File: Server/src/main/java/Server/core/game/node/entity/player/link/quest/QuestRewardComponentItem.java
package core.game.node.entity.player.link.quest;

public class QuestRewardComponentItem {
    int itemId;
    int itemAmount;
    int zoom;

    /**
     *
     * @param itemId The {@link core.game.node.item.Item} id
     * @param zoom The amount to zoom. Doesn't seem to affect anything
     */
    public QuestRewardComponentItem(int itemId, int zoom) {
        this.itemId = itemId;
        this.itemAmount = 1;
        this.zoom = zoom;
    }

    /**
     *
     * @param itemId The {@link core.game.node.item.Item} id
     * @param itemAmount The item's amount
     * @param zoom The amount to zoom. Doesn't seem to affect anything
     */
    public QuestRewardComponentItem(int itemId, int itemAmount, int zoom) {
        this.itemId = itemId;
        this.itemAmount = itemAmount;
        this.zoom = zoom;
    }
}

File: Server/src/main/java/Server/core/game/node/entity/player/link/quest/QuestRewardComponentItem.kt
package core.game.node.entity.player.link.quest

class QuestRewardComponentItem {

    /**
     * [core.game.node.item.Item] id
     */
    @JvmField
    var itemId: Int

    /**
     * Item amount property
     */
    @JvmField
    var itemAmount: Int

    /**
     * Requirement type
     */
    @JvmField
    var zoom: Int

    /**
     *
     * @param itemId The [core.game.node.item.Item] id
     * @param zoom The amount to zoom. Doesn't seem to affect anything
     */
    constructor(itemId: Int, zoom: Int) {
        this.itemId = itemId
        this.itemAmount = 1
        this.zoom = zoom
    }

    /**
     *
     * @param itemId The [core.game.node.item.Item] id
     * @param itemAmount The item's amount
     * @param zoom The amount to zoom. Doesn't seem to affect anything
     */
    constructor(itemId: Int, itemAmount: Int, zoom: Int) {
        this.itemId = itemId
        this.itemAmount = itemAmount
        this.zoom = zoom
    }
}
